src/errors.rs

```diff
@@ -88,3 +88,23 @@ pub enum MPCError {
         bad_shares: Vec<usize>,
     },
 }
+
+/// Represents an error during the proving or verifying of a constraint system.
+#[derive(Fail, Clone, Debug, Eq, PartialEq)]
+pub enum R1CSError {
+    // TODO: make better errors
+    #[fail(display = "Invalid R1CS construction.")]
+    InvalidR1CSConstruction,
+    // When trying to access a variable that has an Err as its value assignment
+    #[fail(display = "Variable does not have a value assignment.")]
+    InvalidVariableAssignment,
```

Comment:
Is the InvalidVariableAssignment ever used aside from the case where the verifier doesn't have the witness data to fill in?

If it isn't, should we rename it to be MissingAssignment?

If it is, should we split the error types?

Reply:
It's only used when the verifier doesn't have witness data.
Sure, we can rename it to MissingAssignment!